File: joonas/035.cpp
class Solution {
public:
    int maxIncreaseKeepingSkyline(vector<vector<int>>& grid) {
        int n = grid.size();
        int sum = 0, total = 0;
        vector<int> rs(n, 0), cs(n, 0);
        for(int r=0; r<n; ++r){
            for(int c=0; c<n; ++c) {
                rs[r] = max(rs[r], grid[r][c]);
                total += grid[r][c];
            }
        }
        for(int r=0; r<n; ++r){
            for(int c=0; c<n; ++c) cs[c] = max(cs[c], grid[r][c]);
        }
        for(int r=0; r<n; ++r){
            for(int c=0; c<n; ++c) {
                sum += min(rs[r], cs[c]);
            }
        }
        return sum - total;
    }
};

File: jlogkim/Season2/035.java
class Solution {
    public int maxIncreaseKeepingSkyline(int[][] grid) {
        int max_col[] = new int[grid.length];
        int max_row[] = new int[grid[0].length];
        
        int sum=0;
        for(int r=0;r<grid.length;r++){
            for(int c=0;c<grid[0].length;c++){
                if(grid[r][c]>max_col[c])max_col[c]=grid[r][c];
                if(grid[r][c]>max_row[r])max_row[r]=grid[r][c];
            }
        }
        int min;
        for(int r=0;r<grid.length;r++){
            for(int c=0;c<grid[0].length;c++){
                min=Math.min(max_col[c],max_row[r]);
                if(min>grid[r][c])sum+=(min-grid[r][c]);
            }
        }
        return sum;
    }
}